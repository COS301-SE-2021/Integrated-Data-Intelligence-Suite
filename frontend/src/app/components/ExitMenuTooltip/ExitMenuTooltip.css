#exit_menu{
      border: 1px solid black;
    border-radius: 0.7em;
    background-color: #000000;
    color: white;
}


/*The little arrow in the tooltip*/
div.ant-dropdown.ant-dropdown-show-arrow.ant-dropdown-placement-bottomRight>div.ant-dropdown-arrow{
    top: 6px;
    border-top-color: #000000 !important;
    border-right-color: #000000 !important;
    border-bottom-color: #000000 !important;
    border-left-color: #000000 !important;
    /* box-shadow: -2px -2px 5px rgba(0, 0, 0, 0.06); */
}

.ant-dropdown-placement-topCenter >
.ant-dropdown-arrow, .ant-dropdown-placement-topLeft >
.ant-dropdown-arrow, .ant-dropdown-placement-topRight >
.ant-dropdown-arrow {
    border-right-color: black;
    border-bottom-color: black;
    border-left-color: black;
    border-top-color: black;
}

/*The LINKs in the tooltip*/
.logout_link{
    color: white !important;
}
