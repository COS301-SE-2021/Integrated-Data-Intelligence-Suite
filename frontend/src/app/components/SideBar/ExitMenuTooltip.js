import React, {Component} from "react";
import {Menu} from "antd";
import {LockOutlined, PoweroffOutlined} from "@ant-design/icons";
import {
    BrowserRouter as Router,
    Switch,
    Route, Link
} from "react-router-dom";

class ExitMenuTooltip extends React.Component {
    render() {
        return (
            <>
                <Router>
                    <Menu id={'exit_menu'}>
                        <Menu.Item key="1" icon={<LockOutlined/>}>
                            Lock
                        </Menu.Item>
                        <Menu.Item key="2" icon={<PoweroffOutlined/>}>
                            <Link to="/login">logout</Link>
                        </Menu.Item>
                    </Menu>
                </Router>
            </>
        );
    }
}

export default ExitMenuTooltip;