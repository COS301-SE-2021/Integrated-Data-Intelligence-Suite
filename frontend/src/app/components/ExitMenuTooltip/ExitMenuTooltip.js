import React, {Component} from "react";
import {Menu} from "antd";
import {LockOutlined, PoweroffOutlined} from "@ant-design/icons";
import './ExitMenuTooltip.css';
import {
    Switch,
    Route, Link
} from "react-router-dom";

class ExitMenuTooltip extends React.Component {
    render() {
        return (
            <>
                <Menu id={'exit_menu'}>
                    <Menu.Item key="2" icon={<PoweroffOutlined/>}>
                        <Link className={'logout_link'} to="/login">logout</Link>
                    </Menu.Item>
                </Menu>

            </>
        );
    }
}

export default ExitMenuTooltip;