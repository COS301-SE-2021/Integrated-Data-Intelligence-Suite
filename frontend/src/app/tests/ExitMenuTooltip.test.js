import React from 'react';
import renderer from 'react-test-renderer';
import ExitMenuTooltip from "../components/ExitMenuTooltip/ExitMenuTooltip";
import {BrowserRouter} from "react-router-dom";

it('ExitMenuTooltip renders correctly', () => {
    const tree = renderer
        .create(
            <BrowserRouter>
                <ExitMenuTooltip/>
            </BrowserRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});