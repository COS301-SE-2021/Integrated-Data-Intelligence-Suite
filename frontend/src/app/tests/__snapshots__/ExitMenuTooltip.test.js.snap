// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExitMenuTooltip renders correctly 1`] = `
Array [
  <ul
    className="ant-menu ant-menu-root ant-menu-vertical ant-menu-light"
    data-menu-list={true}
    id="exit_menu"
    onKeyDown={[Function]}
    role="menu"
    tabIndex={0}
  >
    <li
      className="ant-menu-item"
      data-menu-id="exit_menu-1"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
      tabIndex={-1}
    >
      <span
        aria-label="lock"
        className="anticon anticon-lock ant-menu-item-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="lock"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
          />
        </svg>
      </span>
      <span
        className="ant-menu-title-content"
      >
        Lock
      </span>
    </li>
    <li
      className="ant-menu-item"
      data-menu-id="exit_menu-2"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
      tabIndex={-1}
    >
      <span
        aria-label="poweroff"
        className="anticon anticon-poweroff ant-menu-item-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="poweroff"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M705.6 124.9a8 8 0 00-11.6 7.2v64.2c0 5.5 2.9 10.6 7.5 13.6a352.2 352.2 0 0162.2 49.8c32.7 32.8 58.4 70.9 76.3 113.3a355 355 0 0127.9 138.7c0 48.1-9.4 94.8-27.9 138.7a355.92 355.92 0 01-76.3 113.3 353.06 353.06 0 01-113.2 76.4c-43.8 18.6-90.5 28-138.5 28s-94.7-9.4-138.5-28a353.06 353.06 0 01-113.2-76.4A355.92 355.92 0 01184 650.4a355 355 0 01-27.9-138.7c0-48.1 9.4-94.8 27.9-138.7 17.9-42.4 43.6-80.5 76.3-113.3 19-19 39.8-35.6 62.2-49.8 4.7-2.9 7.5-8.1 7.5-13.6V132c0-6-6.3-9.8-11.6-7.2C178.5 195.2 82 339.3 80 506.3 77.2 745.1 272.5 943.5 511.2 944c239 .5 432.8-193.3 432.8-432.4 0-169.2-97-315.7-238.4-386.7zM480 560h64c4.4 0 8-3.6 8-8V88c0-4.4-3.6-8-8-8h-64c-4.4 0-8 3.6-8 8v464c0 4.4 3.6 8 8 8z"
          />
        </svg>
      </span>
      <span
        className="ant-menu-title-content"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          logout
        </a>
      </span>
    </li>
  </ul>,
  <div
    aria-hidden={true}
    style={
      Object {
        "display": "none",
      }
    }
  />,
]
`;
